src/NovelRT/Atom.cpp

```diff
@@ -5,45 +5,86 @@
 
 namespace NovelRT
 {
-    Atom Atom::getNextEventHandlerId() noexcept
+    Atom Atom::GetNextEventHandlerId() noexcept
     {
         static std::atomic_uintptr_t _nextEventHandlerId(0);
         auto value = ++_nextEventHandlerId;
         return Atom(value);
     }
 
-    Atom Atom::getNextFontSetId() noexcept
+    Atom Atom::GetNextFontSetId() noexcept
     {
         static std::atomic_uintptr_t _nextFontSetId(0);
         auto value = ++_nextFontSetId;
         return Atom(value);
     }
 
-    Atom Atom::getNextTextureId() noexcept
+    Atom Atom::GetNextTextureId() noexcept
     {
         static std::atomic_uintptr_t _nextTextureId(0);
         auto value = ++_nextTextureId;
         return Atom(value);
     }
 
-    Atom Atom::getNextComponentTypeId() noexcept
+    Atom Atom::GetNextComponentTypeId() noexcept
     {
         static std::atomic_uintptr_t _nextComponentTypeId(0);
         auto value = ++_nextComponentTypeId;
         return Atom(value);
     }
 
-    Atom Atom::getNextEntityId() noexcept
+    Atom Atom::GetNextEntityId() noexcept
     {
         static std::atomic_uintptr_t _nextEntityId(0);
         auto value = ++_nextEntityId;
+
+        if (value == std::numeric_limits<uintptr_t>::max())
+        {
+            value = ++_nextEntityId;
+        }
+
         return Atom(value);
     }
 
-    Atom Atom::getNextSystemId() noexcept
+    Atom Atom::GetNextSystemId() noexcept
     {
         static std::atomic_uintptr_t _nextSystemId(0);
         auto value = ++_nextSystemId;
         return Atom(value);
     }
-} // namespace NovelRT
+
+    Atom Atom::GetNextEcsTextureId() noexcept
+    {
+        static std::atomic_uintptr_t _nextEcsTextureId(0);
+        auto value = ++_nextEcsTextureId;
+        return Atom(value);
+    }
+
+    Atom Atom::GetNextEcsVertexDataId() noexcept
+    {
+        static std::atomic_uintptr_t _nextEcsVertexDataId(0);
+        auto value = ++_nextEcsVertexDataId;
+        return Atom(value);
+    }
+
+    Atom Atom::GetNextEcsPrimitiveId() noexcept
+    {
+        static std::atomic_uintptr_t _nextEcsPrimitiveId(0);
+        auto value = ++_nextEcsPrimitiveId;
+        return Atom(value);
+    }
+
+    Atom Atom::GetNextEcsGraphicsPipelineId() noexcept
+    {
+        static std::atomic_uintptr_t _nextEcsGraphicsPipelineId(0);
+        auto value = ++_nextEcsGraphicsPipelineId;
+        return Atom(value);
+    }
+
+    Atom Atom::GetNextEcsPrimitiveInfoConfigurationId() noexcept
+    {
+        static std::atomic_uintptr_t _nextEcsPrimitiveInfoConfigurationId(0);
+        auto value = ++_nextEcsPrimitiveInfoConfigurationId;
+        return Atom(value);
+    }
+}
```

Comment:
It seems we lost the `// namespace NovelRT` comment here - is that intentional?

Reply:
we shouldn't have those comments. If we do then its a merge issue.

Reply:
I vaguely remember seeing these comments in other parts of this PR, hence my asking - this one seemed like the exception, rather than the rule.

Reply:
I hate IDEs.

That is all.